src/Core/Domain/Product/Shop/Command/SetProductShopsCommand.php

```diff
@@ -47,23 +43,25 @@ class CopyProductToShopCommand
     private $sourceShopId;
 
     /**
-     * @var ShopId
+     * @var ShopId[]
      */
-    private $targetShopId;
+    private $shopIds;
 
     /**
      * @param int $productId
      * @param int $sourceShopId
-     * @param int $targetShopId
+     * @param int[] $shopIds
```

Comment:
You'll need to detail the purpose of those fields with more details, especially `sourceShopId` which was kind of explicit when the command was `CopyProductToShopCommand` but a bit less intuitive now that the command is `SetProductShopsCommand`

It makes sense to have a source for a copy, but why for setting associations? I don't have a better naming for this parameter though so at least its purpose needs to be detailed a bit more.